Add swap/mount backend supports where needed and correct minor repeating spelling issue

jtpittman195 opened the pull request:
These commits add swap printing support to all available backends, add mount print support to the multipath backend, and correct a repeating typo in the code.

Commit:
ssmlib: backends: add swap support to backends

Just as volumes can be mounted, they can also be used as swap
volumes.  Add support for swap reporting to each available backend.

Signed-off-by: John Pittman <jpittman@redhat.com>

Commit:
multipath: add mount support to multipath backend

Currently the multipath backend is not configured to show the
mount point.  Add that support.

Signed-off-by: John Pittman <jpittman@redhat.com>

Commit:
ssm: correct all misspellings of additional

In several places additional is incorrectly spelled "aditional".
Correct these misspellings.

Signed-off-by: John Pittman <jpittman@redhat.com>

lczerner commented on ssmlib/backends/multipath.py:
```diff
@@ -26,6 +26,7 @@
 __all__ = ["MultipathDevice"]
 
 MP="multipath"
+DM_DEV_DIR = "/dev"
```
DM_DEV_DIR can be set using environment variable. See backend/lvm.py.

lczerner commented on ssmlib/backends/multipath.py:
```diff
@@ -102,9 +104,12 @@ def get_volume_data(self, volname):
+            else:
+                for swap in misc.get_swaps():
+                    if swap[0] == data['dev_name']:
+                        data['mount'] = "SWAP"
```
get_volume_data is called for every volume and so it would make much more sense to get_swaps in init (as you do with get_mounts) and save the results.

jtpittman195 replied:
sure, thanks lukas, i will check ASAP

jtpittman195 replied:
Hi Lukas, sorry for the wait.  The tests have been added now.  Please review at your convenience.  Thanks!

lczerner reviewed (changes requested):
Clearly Jan completely forgot to get the mounts data and it needs to be fixed, so thank you for that.
Could you also add some tests for this ? Unittests at least since bashtests seems to be completely broken and useless for the mutlipath unfortunatelly.

lczerner commented:
I am about to add more comments and review rest of the changes. I am sorry but I am not used to work with github web UI, I usually review email patches only so this is something new and I am trying to give it a chance :)

-Lukas

lczerner commented on man8/ssm.8:
```diff
@@ -106,13 +106,13 @@ show this help message and exit
 .TP
 .B \-v\fP,\fB  \-\-verbose
-Show aditional information while executing.
+Show additional information while executing.
```
You're changing a generated file. In this case there is no reason to change anything else since the changes will be pulled from ssmlib/main.py. See the doc/ for more information on how the documentation is generated in ssm.
Also I think that Jan made a mistake by including the ssm.8 in the git. I should fix it.

jtpittman195 replied:
fixed in latest commits

lczerner commented on ssmlib/backends/crypt.py:
```diff
@@ -197,6 +197,10 @@ def __init__(self, *args, **kwargs):
             dm['real_dev'] = misc.get_real_device(devname)
             if dm['real_dev'] in self.mounts:
                 dm['mount'] = self.mounts[dm['real_dev']]['mp']
+            else:
+                for swap in misc.get_swaps():
+                    if swap[0] == dm['real_dev']:
+                        dm['mount'] = "SWAP"
```
Please move the get_swaps() to the init method of the base class. That way you don't have to call it for every device you're processing. This applies to all the backends you're making this change for

lczerner replied:
I would also like to see some added tests for this.

jtpittman195 replied:
crypt bash tests and multipath bash tests are broken

jtpittman195 replied:
looking further, it seems the crypt bash tests expect luks1 and my system is luks2; will look at this more later

jtpittman195 replied:
crypt bash testing fixed and swap test added

lczerner commented on ssmlib/main.py:
```diff
@@ -454,7 +453,7 @@ def __init__(self, options, data=None):
                 devices['mount'] = mounts[devices['dev_name']]['mp']
 
-        for item in swaps:
+        for item in misc.get_swaps():
```
Yeah, I think it does make sense here since we only access it once.

jtpittman195 commented:
Thanks again Lukas.  I've updated the current commits with your requested changes.  I'll add an additional commit with the requested test additions.  Will let you know once I'm done.

jtpittman195 committed:
unittests: add multipath mount test

Add test verifying mount check routine within get_volume_data()
method from multipath backend.  Also add real_dev listing to
mount_data as it is needed in get_volume_data() check and
fix a newline error.

Signed-off-by: John Pittman <jpittman@redhat.com>

jtpittman195 committed:
unittests: add swap mount tests in available unittests

Add a basic volume swap test to the two available unittests,
lvm and multipath.

Signed-off-by: John Pittman <jpittman@redhat.com>

jtpittman195 committed:
bashtests: add LUKS2 support to crypt bashtests

The check for type was hard coded to LUKS1.  Add support for LUKS2
by adding a helper function to grab the version from luksDump and
implement the function across the individual commands.

Signed-off-by: John Pittman <jpittman@redhat.com>

jtpittman195 committed:
bashtests: add swap check for crypt volumes

Add test for crypt volume initialized as swap.  Also add the swap
commands to the dependency check in test.py.

Signed-off-by: John Pittman <jpittman@redhat.com>

jtpittman195 committed:
bashtests: settle udev after configuring multipath

Since we're not sure when the multipath devices are fully created
along with sub-paths, we need to ensure that udev is settled prior
to exiting mpath_setup().  Otherwise the rest of the script can
become racy. Also add udevadm to binary check as we need this
command.

Signed-off-by: John Pittman <jpittman@redhat.com>

jtpittman195 committed:
bashtests: ignore mpath devices outside testing

If the multipath bashtests are run on a system that already has
multipath devices present, the tests will grab the wrong devices.
Ensure the tests only look at the multipath device we have created.
Also, as the multipath tests require targetcli and iscsiadm, add
those to the system dependencies list and fix a newline error as
well.

Signed-off-by: John Pittman <jpittman@redhat.com>

lczerner commented on ssmlib/backends/crypt.py:
```diff
@@ -197,6 +198,10 @@ def __init__(self, *args, **kwargs):
+            else:
+                for swap in self.swaps:
+                    if swap[0] == dm['real_dev']:
+                        dm['mount'] = "SWAP"
```
There needs to be a break after the assignment. There is no need to iterate over the rest of the entries in swap. This applies to all places in this patch except the one in DeviceInfo of course.

jtpittman195 replied:
Thanks Lukas and sorry for the wait.  Good catch.  I've made the requested correction.  Please feel free to review at your convenience.

lczerner reviewed (changes requested):
Thanks for the work, it looks good. When the small problem I commented on in the first commit is fixed I'll merge it.

Thanks!
-Lukas

jtpittman195 committed:
ssmlib: backends: break after matching swaps

As currently written, when attempting to match swaps, after the
match, the loop will continue to iterate.  This is unneeded.
Correct by breaking from loop once a match is found.

Signed-off-by: John Pittman <jpittman@redhat.com>
